src/core/shape_inference/include/range_shape_inference.hpp

```diff
@@ -11,6 +11,31 @@ namespace op {
 
 namespace ShapeInferRange {
 
+template <class TRShape, typename std::enable_if<std::is_same<TRShape, PartialShape>::value>::type* = nullptr>
+void dynamic_case(const Node* op,
+                  std::vector<TRShape>& output_shapes,
+                  const double& start,
+                  const double& step,
+                  bool start_val,
+                  bool step_val) {
+    output_shapes[0] = ov::PartialShape::dynamic(1);
+    if (op->get_input_size() == 3 && step_val && step == 1) {
+        auto start_symbol = op->input_value(0).get_tensor().get_value_symbol();
+        auto stop_symbol = op->input_value(1).get_tensor().get_value_symbol();
+        if (start_val && start == 0 && !stop_symbol.empty()) {
+            output_shapes[0][0].set_symbol(stop_symbol[0]);
+        }
+    }
+}
+
+template <class TRShape, typename std::enable_if<!std::is_same<TRShape, PartialShape>::value>::type* = nullptr>
+void dynamic_case(const Node* op,
+                  std::vector<TRShape>& output_shapes,
+                  const double& start,
+                  const double& step,
+                  bool start_val,
+                  bool step_val) {}
```

Comment:
should we return error in this case?

Reply:
No, we should not. This shape inference will be executed during the inference in the plugin. Since data is being fed into the parameters of the model, all the shapes are known for sure. We have made an agreement with plugin owners not to add such useless checks since every processor tick matters during the inference.